libs/wingsdk/src/target-sim/schedule.ts

```diff
@@ -0,0 +1,121 @@
+import { Construct } from "constructs";
+import * as cloud from "../cloud";
+import { Code } from "../core";
+import { Function } from "./function";
+import { convertBetweenHandlers } from "../utils/convert";
+import { join } from "path";
+import { EventMapping } from "./event-mapping";
+import { ISimulatorResource } from "./resource";
+import { BaseResourceSchema } from "../testing";
+import { Duration, IInflightHost, Resource } from "../std";
+import { bindSimulatorResource, makeSimulatorJsClient } from "./util";
+import { SCHEDULE_TYPE, ScheduleSchema } from "./schema-resources";
+
+/**
+ * Simulator implementation of `cloud.Schedule`.
+ * 
+ * @inflight `@winglang/sdk.cloud.IScheduleClient`
+ */
+export class Schedule extends cloud.Schedule implements ISimulatorResource {
+  private readonly cronExpression: string;
+  
+  constructor(scope: Construct, id: string, props: cloud.ScheduleProps = {}) {
+    super(scope, id, props)
+    const { rate, cron } = props;
+
+    if (rate && cron) {
+      throw new Error("rate and cron cannot be configured simultaneously.");
+    }
+    if (!rate && !cron) {
+      throw new Error("rate or cron need to be filled.");
+    }
+    if (rate && rate.seconds < 60) {
+      throw new Error("rate can not be set to less than 1 minute.");
+    }
+    if (cron && cron.split(" ").length > 5) {
+      throw new Error(
+        "cron string must be UNIX cron format [minute] [hour] [day of month] [month] [day of week]"
+      );
+    }
+
+    this.cronExpression = cron ?? this.convertDurationToCron(rate!);
+  }
+
+  // helper function to convert duration to a cron string
+  // maybe this belongs in a util library but for now it's here
+  private convertDurationToCron(dur: Duration): string {
```

Comment:
Will this work for sub-minute durations? e.g. what if I set a schedule of every 10 seconds? Or every 2.5 seconds? Any cases that aren't supported we should probably throw an error

Reply:
great catch I added some tests and exceptions for anything that would result in sub minute precision.